### PR TITLE
[CP-1497] Fixing MuditaOS release separation in dev mode

### PR DESCRIPTION
**Pull request by igorbezsmertnyi**
Jira: [CP-1497]

**Description**
Fix release separation in dev mode
Reflector release retrieving logic

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-1497]: https://appnroll.atlassian.net/browse/CP-1497?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/update/helpers/version-formatter.helper.test.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+describe("Release candidate release", () => {
+  test("returns provided version", () => {
+    expect(versionFormatter("1.2.3-rc.1")).toEqual("1.2.3-rc.1")
+  })
+})
+
+describe("Daily release", () => {
+  test("returns formatted version", () => {
+    expect(versionFormatter("1.2.3-daily.2022.08.12")).toEqual(
+      "1.2.3-daily.2022.8.12"
```
Comment (dkarski) [inline packages/app/src/update/helpers/version-formatter.helper.test.ts:23]: This formatter logic handles the issue on the OS CI side, and daily will be supported after marge? 🤔 
https://appnroll.atlassian.net/browse/CP-1003

File: packages/app/src/update/helpers/version-formatter.helper.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  if (releaseSuffixArray.length < 4) {
+    return version
+  }
+
+  // AUTO DISABLED - fix me if you like :)
+  // eslint-disable-next-line prefer-const
+  let [name, year, month, day] = releaseSuffixArray
+  month = month?.replace(/0/g, "")
+  day = day?.replace(/0/g, "")
```
Comment (dkarski) [inline packages/app/src/update/helpers/version-formatter.helper.ts:23]: Remove the lint rule or refactor a code 😉 
```suggestion
  const [name, year, m, d] = releaseSuffixArray
  const month = m?.replace(/0/g, "")
  const day = d?.replace(/0/g, "")
```

File: packages/app/src/update/presenters/github-release.presenter.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+export class GithubReleasePresenter {
+  static toRelease(item: ReleaseManifest): Release {
+    const version = versionFormatter(item.version)
+    const releaseType = prerelease(version)
+
+    return {
+      version,
+      type: releaseType
+        ? (releaseType[0] as ReleaseType)
```
Comment (dkarski) [inline packages/app/src/update/presenters/github-release.presenter.ts:19]: Please avoid type assertion if possible, to avoid application crashes.

**Review by wercix — DISMISSED** [review]

File: packages/app/src/update/helpers/version-formatter.helper.test.ts (hunk 1 above)
Comment (igorbezsmertnyi) [reply packages/app/src/update/helpers/version-formatter.helper.test.ts:23]: Yes

File: packages/app/src/update/helpers/version-formatter.helper.ts (hunk 2 above)
Comment (igorbezsmertnyi) [reply packages/app/src/update/helpers/version-formatter.helper.ts]: https://github.com/mudita/mudita-center/pull/979/commits/ee21f435f9bab9e418666e907b7c3198820fde9f

**Review by dkarski — APPROVED** [review]

File: packages/app/src/update/helpers/version-formatter.helper.test.ts (hunk 1 above)
Comment (dkarski) [reply packages/app/src/update/helpers/version-formatter.helper.test.ts:23]: So... What do you think to take this issue to the active sprint and pass to tests with this issue? 🤔

**Review by wercix — APPROVED** [review]

**Pull request merged**